### PR TITLE
DB-10537 Fix imports for build

### PR DESCRIPTION
**Pull request by jpanko1**
These changes are included in PR https://github.com/splicemachine/splice-machine-spark-connector/pull/23 for DB-9924 in commit https://github.com/splicemachine/splice-machine-spark-connector/pull/23/commits/5528e45c8d450369ef066dc04433bdc76dd81110

### CODE REVIEW

File: spark2.3/build-assembly.sbt [hunk 1]
```diff
@@ -55,6 +56,7 @@ assemblyExcludedJars in assembly := {
     // Included in / Conflicts with
     // jcl-over-slf4j-1.7.5.jar
     "commons-logging-1.1.1.jar",
+    "commons-lang3-3.6.jar",
```
Comment (jaceklaskowski) [inline spark2.3/build-assembly.sbt:59]: How does this help "fix imports for build"? Why is this needed?

File: spark2.3/build-assembly.sbt [hunk 2]
```diff
@@ -19,6 +19,7 @@ assemblyMergeStrategy in assembly := {
   case PathList("META-INF", "io.netty.versions.properties")  => MergeStrategy.first
   // db-engine-2.8.0.1926.jar VS splice_encoding-2.8.0.1926.jar
   case PathList("com", "splicemachine", "utils", _*)  => MergeStrategy.first
+  case PathList("net", "jcip", "annotations", "GuardedBy.class") => MergeStrategy.last
```
Comment (jaceklaskowski) [inline spark2.3/build-assembly.sbt:22]: Why is this needed to "fix imports for build"?

File: spark2.3/src/main/scala/splice/SpliceDataSource.scala [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package splice
 
-import com.splicemachine.spark2.splicemachine.SplicemachineContext
+import _root_.com.splicemachine.spark2.splicemachine.SplicemachineContext
```
Comment (jaceklaskowski) [inline spark2.3/src/main/scala/splice/SpliceDataSource.scala:3]: Why is this needed? It seemed working fine till now.

File: spark2.4/build-assembly.sbt [hunk 4]
```diff
@@ -56,6 +56,7 @@ assemblyExcludedJars in assembly := {
     // Included in / Conflicts with
     // jcl-over-slf4j-1.7.5.jar
     "commons-logging-1.1.1.jar",
+    "commons-lang3-3.7.jar",
```
Comment (jaceklaskowski) [inline spark2.4/build-assembly.sbt:59]: Is this the only change for spark2.4 (compared to spark2.3)?

NB With these duplications I think developing a sbt [AutoPlugin](https://www.scala-sbt.org/1.x/api/sbt/AutoPlugin.html) would make it sharable across builds.

File: spark2.3/src/main/scala/splice/SpliceDataSource.scala (hunk 3 above)
Comment (jpanko1) [reply spark2.3/src/main/scala/splice/SpliceDataSource.scala:3]: sbt started giving some errors like 

`splice-machine-spark-connector/spark2.3/src/main/scala/splice/SpliceDataSource.scala:3:12: object splicemachine is not a member of package splice.com`
`[error] import com.splicemachine.spark2.splicemachine.SplicemachineContext`

apparently as a result of new shading in jars in the spliceengine project.

File: spark2.3/build-assembly.sbt (hunk 2 above)
Comment (jpanko1) [reply spark2.3/build-assembly.sbt:22]: This was done just to sync the spark2.3 code with [the same change previously done in the spark2.4 code](https://github.com/splicemachine/splice-machine-spark-connector/commit/b01c0b86400ffb7fbdffcae56dce68778e37f774).

File: spark2.3/build-assembly.sbt (hunk 1 above)
Comment (jpanko1) [reply spark2.3/build-assembly.sbt:59]: `sbt assembly` started giving errors like

`[error] deduplicate: different file contents found in the following:`
`[error] ~/.ivy2/cache/org.apache.commons/commons-lang3/jars/commons-lang3-3.6.jar:org/apache/commons/lang3/tuple/Triple.class`
`[error] ~/.m2/repository/com/splicemachine/utilities/3.2.0.1982-SNAPSHOT/utilities-3.2.0.1982-SNAPSHOT.jar:org/apache/commons/lang3/tuple/Triple.class`

apparently as a result of new shading in jars in the spliceengine project.

File: spark2.3/src/main/scala/splice/SpliceDataSource.scala (hunk 3 above)
Comment (jaceklaskowski) [reply spark2.3/src/main/scala/splice/SpliceDataSource.scala:3]: Do you have the commit perhaps? It'd be nice to have a look (and reference back for the future).

File: spark2.4/build-assembly.sbt (hunk 4 above)
Comment (jpanko1) [reply spark2.4/build-assembly.sbt:59]: In the spark2.4 code, the imports in SpliceDataSource and SpliceRelation were also changed like those in the spark2.3 code.

For AutoPlugin development, wrote [a Jira](https://splicemachine.atlassian.net/browse/DB-10585).

File: spark2.3/src/main/scala/splice/SpliceDataSource.scala (hunk 3 above)
Comment (jpanko1) [reply spark2.3/src/main/scala/splice/SpliceDataSource.scala:3]: Sure, please see https://github.com/splicemachine/spliceengine/commit/56fa7d67695bd73fedebaee59acbeae94fd61e93#diff-939ad65c915560fb7f81cf325a7a2520f74277f01ccd76661b3da46d0e993265

**Review by dgomezferro — APPROVED** [review]

**Review by arnaud-splice — APPROVED** [review]

**Review by carolp-503 — COMMENTED** [review]
+1
